pandas/core/computation/engines.py

```diff
@@ -120,12 +114,8 @@ def _evaluate(self):
             _check_ne_builtin_clash(self.expr)
             return ne.evaluate(s, local_dict=scope)
         except KeyError as e:
-            # python 3 compat kludge
-            try:
-                msg = e.message
-            except AttributeError:
-                msg = str(e)
-            raise UndefinedVariableError(msg)
+            name = str(e)
+            raise UndefinedVariableError(name=name)
```

Comment:
cc @jbrockmendel 

if this path was tested (and reachable) it would have raised `__init__() missing 1 required positional argument: 'is_local'`

fix here is to make `is_local` optional keyword.

Reply:
Thanks for checking this out.  Does this merit a separate PR?

Reply:
> Does this merit a separate PR?

will do. 

on master mypy --check-untyped-defs (with mypy 0.720) reports 372 errors. 304 of these still need to be investigated!

so not yet in a position to be able to prioritize fixes.

Reply:
Also, there was another recent case where there was an un-tested exception we were calling incorrectly.  Will typing prevent that?

Reply:
with check-untyped-defs set (and the relevant module not blacklisted) then in theory, yes.

otherwise, if any type annotations are added to the calling function/method then the body of the function will be checked, so should also report errors.

so check-untyped-defs can serve two purposes, checking functions while still in the process of add type annotations and also checking new functions added without type annotations.

the second scenario would be unnecessary if we used disallow-untyped-defs. That's along way off though.